docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,538 @@
+          "kernel.configuration": [
+            "CONFIG_DRM=y",
+            "CONFIG_DRM_I915=y",
+            "CONFIG_DRM_I915_USERPTR=y",
+            "CONFIG_DRM_I915_GVT=m",
+            "CONFIG_DRM_I915_GVT_KVMGT=m"
+          ]
+        }
+      },
+       {
+        "id": "nvidiaGPU",
+        "domain": "org.opencontainers",
+        "attributes": {
+          "hardware.pci.class-id": "0380",
+          "hardware.pci.vendor-id": "10de",
+          "kernel.configuration": [
+            "CONFIG_HOTPLUG_PCI_PCIE=y",
+            "CONFIG_MODULES=y",
+            "CONFIG_MODULE_UNLOAD=y",
+            "CONFIG_PCI_MMCONFIG=m",
+            "CONFIG_DRM_NOUVEAU=n"
+          ]
+        }
+      }
+    ],
+    "relations": {
+      "graphs": {
+        "intel": {
+          "edges": [
+            {
+              "from": "intelCpu",
+              "to": {
+                "all": ["vfio"]
+              }
+            },
+            {
+              "from": "vfio",
+              "to": {
+                "oneOf": ["intelGpu", "nvidiaGpu"]
+              }
+            }
+          ]
+        },
+        "amd": {
+          "annotations": {
+            "documenation.url": "https://example.com"
+          },
+          "edges": [
+            {
+              "from": "amdCpu",
+              "to": {
+                "all": ["vfio", "nvidiaGpu"]
+              }
+            }
+          ]
+        }
+      },
+      "validationCriteria": [
+        {
+          "graphs": ["intel", "amd"],
+          "condition": "all"
+        }
+      ]
+    }
+  }
+}
+```
+
+## OCI Compatibility Tool
+
+The compatibility tool maintained by OCI should be very minimal in scope
+and it should export pkgs for external tools so they can cover their own use cases.
+
+The suggested scope of functionality should be as follows:
+
+- Create artifact.
+- Pull and push artifact.
+- Validate compatibility specification.
+- Validate host.
+- Output the compatibility spec in a readable form.
+
+### Implementation
+
+The tool should be written in Golang and implement in-tree plugins architecture.
+
+The structure should be similar to:
+
+```console
+cmd/
+pkg/
+  cmd/
+  plugins/
+    core/
+    community/
+      org.telco/
+      org.supercontainers/
+      ...
+  schema/
+```
+
+- `cmd/` - provided cmds.
+- `pkg/cmd` - all cmds should be written in a form that they can be exported.
+- `pkg/plugins` - plugins extracts and validates information on the host.
+  They must match compatibility domains from the compatibility spec.
+  For instance `org.opencontainers`.
+- `pkg/plugins/core` - all core plugins maintained by OCI under `org.opencontainers` domain.
+- `pkg/plugins/community` - all community plugins with their own domains with the schema.
+  Community maintainers will have access to their own domain directory.
+- `pkg/schema` - the spec schema.
+
+### Core Plugins
+
+The following subjects should be added to core plugins list that are maintained by OCI group:
```

Comment:
I disagree with this statement - I think even these top level categories should not be our responsibility to maintain, but a group with expertise. E.g., even node feature discovery would do a good job to cover these categories, and our work would be to ensure that logic is moved outside of their kubernetes project tree and adding in respective communities / community members that have the expertise for each area across platforms, etc.

Reply:
And it's not to say that these groups aren't important, but that (I think) OCI should be scoped to the specs, and not what is in them, which is going to change over time and require more than a few core folks. I do think core folks from here would be happy/willing to be a part of those efforts!

Reply:
You have a point here. But then, do you want to keep in-tree architecture? I think that would mean.. we have to import all external stuff into our repository (under appropriate org), write something that translates the spec to the input of the stuff we import, exceute and translate the output to something we expect?

Reply:
> Btw. by core folks in that case I meant me and you 😁 as we (so far) were the only ones who volunteered to maintain the tool. The core is context of image compatibility.

okay with me! I love building stuff. <3